app/src/main/java/com/kickstarter/viewmodels/MessagesViewModel.java

```diff
@@ -39,14 +46,20 @@
     /** Emits a boolean that determines if the backing info view should be hidden. */
     Observable<Boolean> backingInfoViewHidden();
 
-    /** Emits the participant name to be displayed. */
-    Observable<String> participantNameTextViewText();
-
     /** Emits a list of messages to be displayed. */
     Observable<List<Message>> messages();
 
+    /** Emits the participant name to be displayed. */
+    Observable<String> participantNameTextViewText();
+
     /** Emits the project name to be displayed. */
     Observable<String> projectNameTextViewText();
+
+    /** Emits an empty string when we want to clear the message edit text. */
+    Observable<String> setEmptyMessageEditText();
+
+    /** Emits when we should show a message error toast. */
+    Observable<Boolean> showMessageErrorToast();
```

Comment:
what's the boolean for?

seems like instead this should be a `String` that emits the error message you plucked from the error envelope via `first(env.errorMessages())` and default to something if that doesn't exist.